staging/src/k8s.io/apiserver/pkg/endpoints/discovery/version.go

```diff
@@ -18,49 +18,59 @@ package discovery
 
 import (
 	"net/http"
-	"strconv"
-	"time"
 
 	restful "github.com/emicklei/go-restful"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/endpoints/handlers/negotiation"
-	"k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	"k8s.io/klog/v2"
 )
 
-type APIResourceLister interface {
-	ListAPIResources() ([]metav1.APIResource, string)
-}
-
-type APIResourceListerFunc func() ([]metav1.APIResource, string)
-
-func (f APIResourceListerFunc) ListAPIResources() ([]metav1.APIResource, string) {
-	return f()
-}
-
 // APIVersionHandler creates a webservice serving the supported resources for the version
 // E.g., such a web service will be registered at /apis/extensions/v1beta1.
 type APIVersionHandler struct {
 	serializer runtime.NegotiatedSerializer
 
-	groupVersion      schema.GroupVersion
-	apiResourceLister APIResourceLister
+	groupVersion schema.GroupVersion
+	resourceList *metav1.APIResourceList
+	hash         string
```

Comment:
These two things feel related and probably should be updated together, can we force the code to instantiate them together?

Reply:
Forget it, this is super small, that's fine.